src/Components/Components/src/InjectAttribute.cs

```diff
@@ -10,4 +10,8 @@ namespace Microsoft.AspNetCore.Components;
 [AttributeUsage(AttributeTargets.Property, AllowMultiple = false, Inherited = true)]
 public sealed class InjectAttribute : Attribute
 {
+    /// <summary>
+    /// Gets or sets the object that specifies the key of the service to inject.
+    /// </summary>
+    public object? Key { get; init; }
```

Comment:
We can constraint this further to a string I think, as I believe the scenarios where you take a non-string value are for things like fallback registrations via `KeyedService.AnyKey`. In the context of the compiler this is going end up something like `@inject "key" ServiceType Property`.

Reply:
Are nonstring keys meant to be supported in non-Blazor cases in general? If so it might be nice to leave the option of using them with `[Inject]` even if you can't use them with `@inject`.

Of course, you'd be limited to compile-time constant values since this is an attribute, so I'm not totally clear on what useful use cases there are for this.

Reply:
Yes, that was the reasoning with just going for string directly

Reply:
Well, this might be ok as object. You could also use an Enum here and we would be precluding that